### PR TITLE
Add code coverage report generation

### PR DESCRIPTION
**Pull request by tomncooper**
This PR add `jacoco` code coverage report generation, which should hopefully fix the Sonar Cloud integration.

### CODE REVIEW

File: flink-sql-runner/pom.xml [hunk 1]
```diff
@@ -58,6 +58,13 @@
             <version>${fabric8.kubernetes-client.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.jacoco</groupId> 
+            <artifactId>jacoco-maven-plugin</artifactId>
+            <version>${jacoco.version}</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (MikeEdgar) [inline flink-sql-runner/pom.xml:66]: Why as a dependency?

File: flink-sql-runner/pom.xml [hunk 2]
```diff
@@ -121,4 +128,38 @@
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+        <id>coverage</id>
```
Comment (MikeEdgar) [inline flink-sql-runner/pom.xml:133]: Also will need `-Pcoverage` for the `mvn` command in `.github/worksflows/integration.yaml`

File: flink-sql-runner/pom.xml (hunk 1 above)
Comment (tomncooper) [reply flink-sql-runner/pom.xml:66]: Ooop, meant to take that out.

**Review by MikeEdgar — APPROVED** [review]
I think it worked. You can see in the action logs [1]. This wasn't present for the older jobs.
```
[INFO] 14:02:49.240 Importing 1 report(s). Turn your logs in debug mode in order to see the exhaustive list.
```

Sonar doesn't report coverage on PRs where no code was changed, like this one.

[1] https://github.com/streamshub/flink-sql/actions/runs/11256594024/job/31298858729

**Review by kornys — APPROVED** [review]

**Pull request merged**